publications/tasks.py

```diff
@@ -1,22 +1,71 @@
 from django_q.models import Schedule
-from datetime import date, timezone
 from publications.models import Publication
-from django.core.mail import send_mail
+import requests
+from bs4 import BeautifulSoup
+import json
+import xml.dom.minidom
+from django.contrib.gis.geos import GEOSGeometry
 
-def new_manuscript_updates():
-    """
-    send email for manuscript updates
-    """
-    one_minute_ago = timezone.now() - timezone.timedelta(minutes=1)
-    new_manuscripts = Publication.objects.filter(
-        date = one_minute_ago
-    )
-    subject = 'New Manuscripts'
-    email = "test@example.com"
-    message =f"""\ Hello, we send you updated manuscripts that you were looking for:    {new_manuscripts}   """
-    send_mail(subject, message, from_email= "optimetageo@gmail.com",recipient_list=[email])
+def harvest_demo():
+    url1 = 'https://service.tib.eu/optimeta/index.php/optimeta/oai/?verb=ListRecords&metadataPrefix=oai_dc'
+    harvest_data(url1)
 
+def get_geom(url):
+    #soup = BeautifulSoup(open(url,encoding="utf8"), "html.parser") # forlocal file
+    req= requests.get(url)
+    soup = BeautifulSoup(req.content, 'html.parser')
+    return parse_html(soup)
+    
+def parse_html(content):    
+    for tag in content.find_all("meta"):
+        if tag.get("name", None) == "DC.SpatialCoverage":
+            data = tag.get("content", None)
+            json_object = json.loads(data)
+            return(json_object)
+        else : 
+            pass # return None     
+
+def harvest_data(url):
+    # url1 = os.path.join(os.getcwd(),'harvesting\\journal_1\\oai_dc.xml') # forlocalfile
+    response = requests.get(url)
+    DOMTree = xml.dom.minidom.parseString(response.content)
+    collection = DOMTree.documentElement
+    articles = collection.getElementsByTagName("dc:identifier")                                            
+    articles_count_in_journal = len(articles)  # number of articles in journal
+    for i in range(articles_count_in_journal):
+        identifier = collection.getElementsByTagName("dc:identifier")
+        identifier_value = identifier[i].firstChild.nodeValue
+        if identifier_value.startswith('https://'):
+            link_value = identifier_value               
+            #get geometry from html
+            #identifier_url = os.path.join(os.getcwd(),'harvesting\\journal_1\\article_01.html')
+            geom = get_geom(link_value)
+            geom_data = geom["features"][0]["geometry"]
+            geom_data_string = json.dumps(geom_data)
+            # preparing geometry data in accordance to geosAPI fields
+            type_geom= {'type': 'GeometryCollection'}
+            geom_content = {"geometries" : [geom_data_string]}
+            type_geom.update(geom_content)
+            geom_data = json.dumps(type_geom)
+            # geometry field accept json object as string
+            geom_object = GEOSGeometry(geom_data) #GeometryCollection object
```

Comment:
This could be something that breaks in the real world, so should handle such edge cases here.

- add test case with invalid geometry
- add test case with invalid JSON